39/yarn2

pyphilia opened the pull request:
close #39

juancarlosfarah commented on package.json:
```diff
@@ -17,7 +17,8 @@
     "start": "microbundle-crl watch --no-compress --format modern,cjs",
-    "prepare": "run-s build",
+    "prepack": "run-s build",
+    "prepare":"yarn prepack",
```
Missing space after `:`.

juancarlosfarah commented on src/items/AppItem.js:
```diff
@@ -16,6 +16,7 @@ export const GET_CONTEXT = 'GET_CONTEXT';
 export const GET_CONTEXT_SUCCEEDED = 'GET_CONTEXT_SUCCEEDED';
 
 const requestApiAccessToken = async ({ id, origin, app, apiHost }) => {
+  // eslint-disable-next-line no-undef
```
Is this because of `fetch`? I think you should be able to tell ESLint that we're in a browser context and it should see `fetch`.

juancarlosfarah commented on src/items/AppItem.js:
```diff
@@ -119,6 +120,7 @@ class AppItem extends Component {
     if (type === GET_CONTEXT) {
       // create/reset channel and
       // Listen for messages on port1
+      // eslint-disable-next-line no-undef
```
Same as above, but for `MessageChannel`.

juancarlosfarah reviewed (approved):
Looks good! Just a couple questions about ESLint.

Just to confirm that we're going directly to Yarn 3, right? Also, not sure why this file (`.yarn/releases/yarn-berry.cjs`) is different than one of the other repos.

juancarlosfarah reviewed (approved):
Looks good! Just a couple questions about ESLint.

Just to confirm that we're going directly to Yarn 3, right? Also, not sure why this file (`.yarn/releases/yarn-berry.cjs`) is different than one of the other repos.

pyphilia commented:
@juancarlosfarah I updated the PR so it works yarn3. I also updated the way tests run (using jest instead of `react-script`). Lots of files have changed because of linter.

juancarlosfarah commented on package.json:
```diff
@@ -6,6 +6,9 @@
     "Juan Carlos Farah"
   ],
+  "workspaces": [
+    "example"
```
Why do we need this?

pyphilia replied:
It actually ease the process, as it `yarn` the workspaces at the same time as the core repository.

juancarlosfarah commented on package.json:
```diff
@@ -39,26 +43,31 @@
     "enzyme-adapter-react-16": "1.15.5",
     "enzyme-to-json": "3.6.1",
-    "eslint": "^6.8.0",
+    "eslint": "^6.6.0",
```
Can we pin this and all the other "unpinned" libraries?

pyphilia replied:
No, it depends on `react-scripts`. I'll see for the other packages. Perhaps it will rather be in a update dep branch.

juancarlosfarah replied:
OK

juancarlosfarah commented on package.json:
```diff
@@ -6,6 +6,9 @@
   "contributors": [
     "Juan Carlos Farah"
```
Please add the contributors.

juancarlosfarah reviewed (changes requested):
Looks great! Just some minor comments. 🎉

juancarlosfarah commented:
Got it.

—
Juan Carlos Farah
Vice President of Research
[Graasp Association](https://graasp.org)

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
On Thursday, August 5th, 2021 at 4:17 PM, Kim Lan Phan Hoang ***@***.***> wrote:

> @pyphilia commented on this pull request.
>
> ---------------------------------------------------------------
>
> In [package.json](https://github.com/graasp/graasp-ui/pull/40#discussion_r683496133):
>
>> @@ -6,6 +6,9 @@
>    "contributors": [
>      "Juan Carlos Farah"
>    ],
> +  "workspaces": [
> +    "example"
>
> It actually ease the process, as it yarn the workspaces at the same time as the core repository.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, [view it on GitHub](https://github.com/graasp/graasp-ui/pull/40#discussion_r683496133), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AANAZNFDKFHAUHY4CBUF5RLT3KMOXANCNFSM5BGS263A).
